scripts/docker/buildpack-deps/Dockerfile.ubuntu2004

```diff
@@ -0,0 +1,63 @@
+# vim:syntax=dockerfile
+#------------------------------------------------------------------------------
+# Dockerfile for building and testing Solidity Compiler on CI
+# Target: Ubuntu 19.04 (Disco Dingo)
+# URL: https://hub.docker.com/r/ethereum/solidity-buildpack-deps
+#
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2016-2019 solidity contributors.
+#------------------------------------------------------------------------------
+FROM buildpack-deps:focal AS base
+LABEL version="16"
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN set -ex; \
+        dist=$(grep DISTRIB_CODENAME /etc/lsb-release | cut -d= -f2); \
+        echo "deb http://ppa.launchpad.net/ethereum/cpp-build-deps/ubuntu $dist main" >> /etc/apt/sources.list ; \
+        apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 1c52189c923f6ca9 ; \
+	apt-get update; \
+	apt-get install -qqy --no-install-recommends \
+		build-essential \
+		software-properties-common \
+		cmake ninja-build \
+		libboost-filesystem-dev libboost-test-dev libboost-system-dev \
+		libboost-program-options-dev \
+		libcvc4-dev libz3-static-dev z3-static jq \
+		; \
+	apt-get install -qy python3-pip python3-sphinx; \
+	pip3 install codecov; \
+	rm -rf /var/lib/apt/lists/*
+
+FROM base AS libraries
+
+# EVMONE
+RUN set -ex; \
+	cd /usr/src; \
+	git clone --branch="v0.9.1" --recurse-submodules https://github.com/ethereum/evmone.git; \
+	cd evmone; \
+	mkdir build; \
+	cd build; \
+	cmake -G Ninja -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_PREFIX="/usr" ..; \
+	ninja; \
+	ninja install/strip; \
+	rm -rf /usr/src/evmone
+
```

Comment:
I removed the Hera build from the original Dockerfile. Not sure if I should include it again. Should I?

Reply:
The idea was to just bring it back for reference, s.t. all images used in CI have a dockerfile here explaining what's in them - so in that sense, since that old image has hera in it, it'd be more accurate to keep it in here, unchanged, but it doesn't really matter that much - if we keep this mechanism for the time being, the very next build *shouldn't* include hera, so also fine to remove it right away :-).

Reply:
I bring it back in any case. So we keep things consistent as they were :)

Reply:
> * Remove Hera.

You mean remove it from the `solidity::test`? Because I already dropped it in the new ubuntu images.

> * Remove Hera.
> 
> * Add all those dependency packages we're installing with `apt install` in `config.yml`. They should really be already included in the base image.
> 
> * Rename `Dockerfile.ubuntu1604.clang.ossfuzz` to `Dockerfile.clang.ossfuzz`. It's not using Ubuntu 16.04 but whatever version the clang is using.

But should all those be done in one follow up PR or maybe 3 quick ones for each of the bullet points?


> Also, what about the `libcln-dev` change we did for 22.04? Is it just missing on newer Ubuntu or is it a general change that we should have on 20.04 too?

Ah, it is because `CVC4` has `CLN` as an [optional feature](https://github.com/CVC4/CVC4-archived/blob/master/INSTALL.md#cln--v13-class-library-for-numbers) but Ubuntu 22.04 ships `CVC4` with it enabled by default while Ubuntu 20.04 uses `GMP` by default. See this comment here: https://github.com/ethereum/solidity/issues/13891#issuecomment-1408227233
So for the newer Ubuntu versions that uses `CVC4` we need `libcln-dev` installed.

Reply:
> You mean remove it from the `solidity::test`? Because I already dropped it in the new ubuntu images.

From the 20.04 image. Daniel said not to remove it in this PR because we want the source to match the restored image. But the follow up could just rebuild the image without hera and then removing this bit would be appropriate.

> But should all those be done in one follow up PR or maybe 3 quick ones for each of the bullet points?

Either is fine. I'd usually suggest keeping PRs as focused as possible but some or even all of them will trigger image rebuild so for these I think it's also fine to keep all those things in a single PR.

> So for the newer Ubuntu versions that uses `CVC4` we need `libcln-dev` installed.

Ah, ok then.